[2024-06-16T20:27:15.898+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-16T20:27:15.972+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.data_cleaning manual__2024-06-16T20:20:35+00:00 [queued]>
[2024-06-16T20:27:15.992+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.data_cleaning manual__2024-06-16T20:20:35+00:00 [queued]>
[2024-06-16T20:27:15.993+0000] {taskinstance.py:2306} INFO - Starting attempt 4 of 6
[2024-06-16T20:27:16.031+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): data_cleaning> on 2024-06-16 20:20:35+00:00
[2024-06-16T20:27:16.043+0000] {standard_task_runner.py:63} INFO - Started process 8299 to run task
[2024-06-16T20:27:16.053+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'data_cleaning', 'manual__2024-06-16T20:20:35+00:00', '--job-id', '861', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmp4lb1qe99']
[2024-06-16T20:27:16.059+0000] {standard_task_runner.py:91} INFO - Job 861: Subtask data_cleaning
[2024-06-16T20:27:16.173+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.data_cleaning manual__2024-06-16T20:20:35+00:00 [running]> on host 333385971b19
[2024-06-16T20:27:17.279+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='data_cleaning' AIRFLOW_CTX_EXECUTION_DATE='2024-06-16T20:20:35+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-16T20:20:35+00:00'
[2024-06-16T20:27:17.282+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-16T20:27:17.284+0000] {data_clean.py:15} INFO - starting data cleaning process
[2024-06-16T20:27:17.347+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/dataload/processing/data_clean.py:17 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-16T20:27:18.027+0000] {data_clean.py:35} INFO - saving cleaned data to csv
[2024-06-16T20:27:18.047+0000] {data_clean.py:39} INFO - successfully saved data to csv
[2024-06-16T20:27:18.054+0000] {python.py:237} INFO - Done. Returned value was: Index(['Entity', 'Code', 'Year', 'Nitrogen_oxide_NOx',
       'Sulphur_dioxide_SO_emissions', 'Carbon_monoxide_CO_emissions',
       'Organic_carbon_OC_emissions',
       'Non_methane_volatile_organic_compounds_NMVOC_emissions',
       'Black_carbon_BC_emissions', 'Ammonia_NH_emissions', 'Unnamed_10'],
      dtype='object')
[2024-06-16T20:27:18.065+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-16T20:27:18.095+0000] {xcom.py:675} ERROR - Object of type Index is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-06-16T20:27:18.098+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'pandas.core.indexes.base.Index'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 486, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3197, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/xcom.py", line 246, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/xcom.py", line 673, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Index is not JSON serializable
[2024-06-16T20:27:18.129+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=data_cleaning, run_id=manual__2024-06-16T20:20:35+00:00, execution_date=20240616T202035, start_date=20240616T202715, end_date=20240616T202718
[2024-06-16T20:27:18.167+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 861 for task data_cleaning (Object of type Index is not JSON serializable; 8299)
[2024-06-16T20:27:18.224+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-16T20:27:18.284+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-16T20:27:18.287+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
